crates/krun-server/src/server.rs

```diff
@@ -46,6 +46,27 @@ fn read_request(mut stream: &TcpStream) -> Result<Launch> {
     }
 }
 
+fn get_stdout_stderr(path: &Path, envs: &HashMap<String, String>) -> Result<(Stdio, Stdio)> {
+    if let Some(filename) = path.file_name() {
+        if let Some(filename) = filename.to_str() {
+            let base = if envs.contains_key("XDG_RUNTIME_DIR") {
+                Path::new(&envs["XDG_RUNTIME_DIR"])
+            } else {
+                Path::new("/tmp")
+            };
+            let ts = SystemTime::now().duration_since(UNIX_EPOCH)?.as_millis();
+            let path_stdout = base.join(format!("krun-{filename}-{ts}.stdout"));
+            let path_stderr = base.join(format!("krun-{filename}-{ts}.stderr"));
+            return Ok((
+                File::create_new(path_stdout)?.into(),
+                File::create_new(path_stderr)?.into(),
+            ));
+        }
+    }
+
+    Ok((Stdio::null(), Stdio::null()))
+}
```

Comment:
```suggestion
fn make_stdout_stderr<P>(command: P, envs: &HashMap<String, String>) -> Result<(Stdio, Stdio)>
where
    P: AsRef<Path>,
{
    let command = command.as_ref();
    let Some(filename) = command.file_name() else {
        return Ok((Stdio::null(), Stdio::null()));
    };
    let filename = filename
        .to_str()
        .context("Failed to process command as it contains invalid UTF-8")?;
    let base = if envs.contains_key("XDG_RUNTIME_DIR") {
        Path::new(&envs["XDG_RUNTIME_DIR"])
    } else {
        Path::new("/tmp")
    };
    let ts = SystemTime::now().duration_since(UNIX_EPOCH)?.as_millis();
    let path_stdout = base.join(format!("krun-{filename}-{ts}.stdout"));
    let path_stderr = base.join(format!("krun-{filename}-{ts}.stderr"));
    Ok((
        File::create_new(path_stdout)?.into(),
        File::create_new(path_stderr)?.into(),
    ))
}
```

Reply:
I'm fine with using a bound trait, but I think if the string conversion fails we should also return `(Stdio::null(), Stdio::null())`. I mean, is not much worse that `Path::file_name()` failing, and we're doing that for it.

Reply:
On second thought, could we not just error out if `file_name` is `None`? In which cases are we expecting that?

Reply:
> Returns `None` if the path terminates in `...`

https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name

I think we can just error.

Reply:
Applied the change, bailing out if file_name is None.